packages/jira-adapter/src/filters/forms/forms_types.ts

```diff
@@ -0,0 +1,201 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { ObjectType, ElemID, BuiltinTypes, CORE_ANNOTATIONS, ListType } from '@salto-io/adapter-api'
+import Joi from 'joi'
+import { elements as adapterElements } from '@salto-io/adapter-components'
+import { createSchemeGuard } from '@salto-io/adapter-utils'
+import { JIRA, FORM_TYPE } from '../../constants'
+
+export type detailedFormResponse = {
+  data: {
+    uuid?: string
+    publish?: string
+    id: number
+    design: {
+      settings: {
+        templateFormUuid?: string
+      }
+      questions: {}
+      sections: {}
+      conditions: {}
+    }
+  }
+}
+
+type formResponse = {
+  id: number
+  name: string
+}
+
+export type formsResponse = {
+  data: formResponse[]
+}
+
+export const FORMS_RESPONSE_SCHEME = Joi.object({
+  data: Joi.array().items(Joi.object({
+  }).unknown(true).required()),
+}).unknown(true).required()
+
+export const DETAILED_FORM_RESPONSE_SCHEME = Joi.object({
+  data: Joi.object({
+    uuid: Joi.string().required(),
+    design: Joi.object({
+    }).unknown(true).required(),
+  }).unknown(true).required(),
+}).unknown(true).required()
+
+
+export const createFormType = (): {
+    formType: ObjectType
+    subTypes: ObjectType[]
+  } => {
+  const FormSubmitType = new ObjectType({
+    elemID: new ElemID(JIRA, 'FormSubmitettings'),
+    fields: {
+      lock: {
+        refType: BuiltinTypes.BOOLEAN,
+      },
+      pdf: {
+        refType: BuiltinTypes.BOOLEAN,
+      },
+    },
+  })
+
+  const FormSettingsType = new ObjectType({
+    elemID: new ElemID(JIRA, 'FormSettings'),
+    fields: {
+      templateId: {
+        refType: BuiltinTypes.NUMBER,
+      },
+      name: {
+        refType: BuiltinTypes.STRING,
+      },
+      submit: {
+        refType: FormSubmitType,
+      },
+      templateFormUuid: {
+        refType: BuiltinTypes.STRING,
+      },
+    },
+  })
+  const attributeContnetLayoutType = new ObjectType({
+    elemID: new ElemID(JIRA, 'attributeContnetLayout'),
+    fields: {
+      localId: {
+        refType: BuiltinTypes.STRING,
+        annotations: { [CORE_ANNOTATIONS.HIDDEN_VALUE]: true },
+      },
+    },
+  })
+  const contentLayoutType = new ObjectType({
+    elemID: new ElemID(JIRA, 'contentLayout'),
+    fields: {
+      type: {
+        refType: BuiltinTypes.STRING,
+      },
+      localId: {
+        refType: new ListType(BuiltinTypes.STRING),
+        annotations: { [CORE_ANNOTATIONS.HIDDEN_VALUE]: true },
+      },
+      attrs: {
+        refType: attributeContnetLayoutType,
+      },
+    },
+  })
+  const FormLayoutItemType = new ObjectType({
```

Comment:
nit: 
```suggestion
  const formLayoutItemType = new ObjectType({
```
(also in a few others)